### PR TITLE
Feature/381 snyk python producer

### LINKED ISSUE #381: create a snyk-python producer

### PR DESCRIPTION
**Pull request by northdpole**
closes #381 and #382

### CODE REVIEW

File: components/producers/snyk-python/task.yaml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        echo "Snyk completed successfully! exitcode $exitCode"
+      fi
+    volumeMounts:
+    - mountPath: /scratch
+      name: scratch
+
+  - name: produce-issues
+    imagePullPolicy: IfNotPresent
+    # image: '{{ default "ghcr.io/ocurity/dracon" .Values.image.registry }}/components/producers/snyk-docker:{{ .Chart.AppVersion }}'
+    image: kind-registry:5000/components/producers/snyk-docker:v0.50.1-dirty
```
Comment (ptzianos) [inline components/producers/snyk-python/task.yaml:60]: this image is a dev leftover

File: components/producers/snyk-python/task.yaml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+      name: scratch
+
+  - name: produce-issues
+    imagePullPolicy: IfNotPresent
+    # image: '{{ default "ghcr.io/ocurity/dracon" .Values.image.registry }}/components/producers/snyk-docker:{{ .Chart.AppVersion }}'
+    image: kind-registry:5000/components/producers/snyk-docker:v0.50.1-dirty   
+
+    command: ["/app/components/producers/snyk-docker/snyk-docker-parser"]
+    args:
+    - "-in=/scratch/snyk.out"
```
Comment (ptzianos) [inline components/producers/snyk-python/task.yaml:64]: we should prefer environment variables instead of the flags

Comment (northdpole) [reply components/producers/snyk-python/task.yaml:64]: producers do not get env vars in this version of the project :-)

File: examples/pipelines/snyk-project/pipelinerun.yaml [hunk 3]
```diff
@@ -7,10 +7,18 @@ spec:
   pipelineRef:
     name: dracon-snyk-project
   params:
+  - name: git-clone-url
+    value: https://github.com/adeyosemanputra/pygoat.git
   - name: producer-snyk-docker-api-key
-    value: ""
+    value: "$snyk-api-key"
```
Comment (ptzianos) [inline examples/pipelines/snyk-project/pipelinerun.yaml:13]: this value looks a bit sus. is it supposed to be just an example?

Comment (northdpole) [reply examples/pipelines/snyk-project/pipelinerun.yaml:13]: it is very much an example and on purpose being a $variable

**Review by ptzianos — APPROVED** [review]

**Pull request merged**